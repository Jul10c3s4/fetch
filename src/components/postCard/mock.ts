//geralmente cria-se um mock quando o componente é estático para que o mock seja usado tanto em tests quanto no próprio componente

export const postCardPropsMock = {
    title: "titulo",
    body: "corpo",
    id: 1,
    photo: "https://dfdfdf"
}