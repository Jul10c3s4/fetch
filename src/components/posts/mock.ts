export const postsCardsPropsMocks = {
    posts:[
      {
        id: 1,
        title: 'title 1',
        body: 'body 1',
        photo: 'https://dfdf'
      },
      {
        id: 2,
        title: 'title 2',
        body: 'body 2',
        photo: 'https2://dfdf'
      },
      {
        id: 3,
        title: 'title 3',
        body: 'body 3',
        photo: '3https://dfdf'
      },
      {
        id: 4,
        title: 'title 4',
        body: 'body 4',
        photo: '4https://dfdf'
      },
    ]
}