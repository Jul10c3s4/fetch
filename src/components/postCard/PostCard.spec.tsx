import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';
import '@testing-library/jest-dom/extend-expect';

const props = postCardPropsMock;
describe('<PostCard>', () => {
  it('should render PostCard correctly', () => {
    const { debug } = render(<PostCard {...props} />);
    //o debug é usado para exibir o componente no terminal
    //debug()
    expect(screen.getByRole('img', { name: 'titulo' })).toBeInTheDocument();
    expect(screen.getByRole('img', { name: 'titulo' })).toHaveAttribute('src', props.photo);
    expect(screen.getByAltText(props.title));
    expect(screen.getByRole('heading', { name: 'titulo' })).toBeInTheDocument();
    expect(screen.getByText('corpo')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    //usado para gerar uma pasta com arquivo que contém um snapshot do componente
    expect(container.firstChild).toMatchSnapshot();
  });
});
