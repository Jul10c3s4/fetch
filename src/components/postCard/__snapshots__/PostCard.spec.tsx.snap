// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="titulo"
    src="https://dfdfdf"
  />
  <div
    class="post-content"
  >
    <h1>
      1
    </h1>
    <h2>
      titulo
    </h2>
    <p>
      corpo
    </p>
  </div>
</div>
`;
